processing/src/main/java/org/apache/druid/query/spec/SpecificSegmentQueryRunner.java

```diff
@@ -64,18 +68,26 @@ public Sequence<T> run(final QueryPlus<T> input, final ResponseContext responseC
         )
     );
 
-    final Query<T> query = queryPlus.getQuery();
+    final boolean setName = input.getQuery().getContextBoolean(CTX_SET_THREAD_NAME, true);
 
-    final Thread currThread = Thread.currentThread();
-    final String currThreadName = currThread.getName();
-    final String newName = query.getType() + "_" + query.getDataSource() + "_" + query.getIntervals();
+    final Query<T> query = queryPlus.getQuery();
 
-    final Sequence<T> baseSequence = doNamed(
-        currThread,
-        currThreadName,
-        newName,
-        () -> base.run(queryPlus, responseContext)
-    );
+    final Thread currThread = setName ? Thread.currentThread() : null;
+    final String currThreadName = setName ? currThread.getName() : null;
```

Comment:
Super Nit: 75:77 we are checking the setName variable thrice.
We can easily move them inside  the block which starts from line 81. 
I think it makes the code easier to read as well.

Reply:
I tried that at first, and then realized there's another usage of these variables at line 164.